### ISSUE TITLE
Chromatographic peak detection failed for all files

### ISSUE THREAD

**Issue by NebuchadnezzaraHardrada** [rietho/IPO#54]
I seem to be having problems with your package. I have looked at the previous issues and no one seems to be running into the same problems as I am.  I have tried changing the parameters, using both "centwave" and "matchedFilter" but still keep getting errors and the plots produced are blank (empty).

library(IPO)
filepath <- "E:/Data/Kundai/RStudio/XCMS/MZXML/KPCI2"
files <- list.files(filepath, pattern = "mzXML", recursive = TRUE, full.names = TRUE)
peakpickingParameters <- getDefaultXcmsSetStartingParams('centWave')
peakpickingParameters$step <- c(0.2, 0.3)
peakpickingParameters$fwhm <- c(40, 50)
#setting only one value for steps therefore this parameter is not optimized
peakpickingParameters$steps <- 2

time.xcmsSet <- system.time({ # measuring time
  resultPeakpicking <- optimizeXcmsSet(files = files, 
                                       params = peakpickingParameters, 
                                       nSlaves = 1, 
                                       subdir = NULL,
                                       plot = TRUE)

starting new DoE with:
fwhm: c(40, 50)
snthresh: c(3, 17)
step: c(0.2, 0.3)
steps: 2
sigma: 0
max: 5
mzdiff: 0
index: FALSE

1
Error in xcms::xcmsSet(files = files, method = "matchedFilter", fwhm = xcmsSetParameters$fwhm[task],  : 
  Chromatographic peak detection failed for all files! The first error was: Error in match.arg(params$method, getOption("BioC")$xcms$findPeaks.methods): 'arg' sollte eines  von '' sein

Error in resultIncreased(history) : 
  No isotopes have been detected, peak picking not optimizable by IPO!
Timing stopped at: 0.31 0.11 71.02

> sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
[4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] IPO_1.4.1                  CAMERA_1.34.0              rsm_2.9                   
 [4] BiocInstaller_1.28.0       ropls_1.10.0               xcms_3.0.0                
 [7] MSnbase_2.4.0              ProtGenerics_1.10.0        BiocParallel_1.12.0       
[10] mzR_2.12.0                 Rcpp_0.12.13               yamss_1.4.0               
[13] SummarizedExperiment_1.8.0 DelayedArray_0.4.1         matrixStats_0.52.2        
[16] Biobase_2.38.0             GenomicRanges_1.30.0       GenomeInfoDb_1.14.0       
[19] IRanges_2.12.0             S4Vectors_0.16.0           BiocGenerics_0.24.0

**Comment by rietho** [comment]
Hi!

First, thanks for using IPO :)

I can't see exactly how you got to this error. What's a problem is, that you chose your parameters from the method `centWave`, while you added parameters for the `matchedFilter`-method (like `fwhm`, `snthresh`, `step` or `steps`). This confuses IPO and let's IPO think that you want to use the `matchedFilter`-method.
What confuses me, is that IPO should catch the error in your case. The following part of your code results in a different error:
```
peakpickingParameters <- getDefaultXcmsSetStartingParams('centWave')
peakpickingParameters$step <- c(0.2, 0.3)
peakpickingParameters$fwhm <- c(40, 50)
#setting only one value for steps therefore this parameter is not optimized
peakpickingParameters$steps <- 2

resultPeakpicking <- optimizeXcmsSet(files = files,
                                     params = peakpickingParameters,
                                     nSlaves = 1,
                                     subdir = NULL,
                                     plot = TRUE)
#+ Error in checkParams(params, quantitative_parameters, qualitative_parameters,  : 
#+   The parameter(s) sigma, max, index is/are missing! Please specify; stopping!
```

If you want to use the `centWave`-method, please only specify according parameters. Please see the help of `getDefaultXcmsSetStartingParams` for more details.

**Comment by NebuchadnezzaraHardrada** [comment]
Thank you for your timely response. I have realized that I had mixed up "centwave" and "matchedFilter" options. This is because, after looking up at the previous issues to ensure that I was not repeating an already solved issue, I ended up mixing the code in the vignette and the one used by S Neumann in the last recently solved issue. The software works as shown in the vignette, thank you for your help and for making the package available.
